R/repair-names.R

```diff
@@ -54,14 +105,77 @@ tidy_names <- function(name, syntactic = FALSE, quiet = FALSE) {
   new_name
 }
 
+#' @export
+#' @rdname name-repair
+set_minimal_names <- function(x) {
+  new_names <- minimal_names(names(x), n = length(x))
+  set_names(x, new_names)
+}
+
+#' @export
+#' @rdname name-repair
+set_valid_names <- function(x, quiet = FALSE) {
+  x <- set_minimal_names(x)
+  new_names <- valid_names(names(x), quiet = quiet)
+  set_names(x, new_names)
+}
+
+#' @export
+#' @rdname name-repair
+set_tidy_names <- function(x, quiet = FALSE) {
+  x <- set_minimal_names(x)
+  new_names <- tidy_names(names(x), quiet = quiet)
+  set_names(x, new_names)
+}
+
+#' @export
+#' @rdname name-repair
+check_minimal <- function(name) {
+  if (is.null(name)) {
+    abort(error_names_must_be_non_null())
+  }
+  name
```

Comment:
This function would be called for its side effects, if at all we should return invisibly (here and below).